bon-macros/src/builder/builder_gen/member/config/getter.rs

```diff
@@ -0,0 +1,99 @@
+use std::ops::Deref;
+
+use darling::FromMeta;
+use proc_macro2::Span;
+use syn::spanned::Spanned;
+
+use super::{docs_utils::parse_docs, SpannedKey};
+
+/// Wrapper around the getter config that allows it to work as both a flag and
+/// a config.
+#[derive(Debug)]
+pub(crate) struct OptionalGetterConfig {
+    span: Span,
+    getter_config: Option<GetterConfig>,
+}
```

Comment:
I think you could use `Option<SpannedKey<GetterConfig>>`, where `GetterConfig` is a struct with `name`, `vis` and `docs`. If none of the `name`, `vis` and `doc` were specified (`from_word`), then all those 3 fields would just be `None` (just derive `Default`), so there is no need for a special `Inferred` value or additional `Option` wrapper.

The `SpannedKey` type will contain the span of the `getter` key no matter if it's a simple `getter` or `getter(...)`

Reply:
Something like this will work. The trick is to manually implement `FromMeta` by using an inner struct that derives `FromMeta` (this trick was used in other places in `bon`, you can find it by "struct Parsed"):

```
#[derive(Default, Debug)]
struct GetterConfig {
    name: Option<SpannedKey<syn::Ident>>,
    vis: Option<SpannedKey<syn::Visibility>>,
    docs: Option<SpannedKey<Vec<syn::Attribute>>>,
}

impl FromMeta for GetterConfig {
    fn from_meta(meta: &syn::Meta) -> Result<Self> {
        if let syn::Meta::Path(_) = meta {
            return Ok(Self::default());
        }

        // Reject empty parens such as `#[builder(getter())]`
        crate::parsing::require_non_empty_paren_meta_list_or_name_value(meta)?;

        // Nested `Parsed` struct used as a helper for parsing the verbose form
        #[derive(FromMeta)]
        struct Parsed {
            name: Option<SpannedKey<syn::Ident>>,
            vis: Option<SpannedKey<syn::Visibility>>,

            #[darling(rename = "doc", default, with = parse_docs, map = Some)]
            docs: Option<SpannedKey<Vec<syn::Attribute>>>,
        }

        let Parsed { name, vis, docs } = Parsed::from_meta(meta)?;

        Ok(Self { name, vis, docs })
    }
}

fn parse_docs(meta: &syn::Meta) -> Result<SpannedKey<Vec<syn::Attribute>>> {
    crate::parsing::parse_docs_without_self_mentions("builder struct's impl block", meta)
}
```